src/modules/mahjong/PointValidator/PointValidator.ts

```diff
@@ -211,6 +215,66 @@ class PointValidator {
     return HKHandMapper.INVALID;
   };
 
+  public static validateFlower = (playerFlower: number, bonusTiles: Tile[]): number => {
+    const flowerMapper: { [index: string]: number } = {
+      '1_FLOWER': 1,
+      '1_SEASON': 1,
+      '2_FLOWER': 2,
+      '2_SEASON': 2,
+      '3_FLOWER': 3,
+      '3_SEASON': 3,
+      '4_FLOWER': 4,
+      '4_SEASON': 4,
+    };
+
+    let points = 0;
+
+    bonusTiles.forEach((t) => {
+      const tileStr = t.toString();
+      if (isBonusTileUtils(tileStr)) {
+        if (flowerMapper[tileStr] === playerFlower) points += HKHandMapper.FLOWER.points;
+      }
+    });
+
+    return points;
+  };
+
+  public static validateDragon = (melds: Meld[]): number => {
+    const dragons = [
+      HonorTileTypes.GREENDRAGON.valueOf(),
+      HonorTileTypes.REDDRAGON.valueOf(),
+      HonorTileTypes.WHITEDRAGON.valueOf(),
+    ];
+    let points = 0;
+
+    melds.forEach((meld) => {
+      if (meld.type === MeldEnums.TRIPLET || meld.type === MeldEnums.QUAD) {
+        if (dragons.includes(meld.tiles[0])) points += HKHandMapper.DRAGON_MELD.points;
+      }
+    });
+
+    return points;
+  };
+
+  public static validateWind(melds: Meld[], wind: WindEnums): number {
+    const winds = [
+      HonorTileTypes.EAST.valueOf(),
+      HonorTileTypes.SOUTH.valueOf(),
+      HonorTileTypes.WEST.valueOf(),
+      HonorTileTypes.NORTH.valueOf(),
+    ];
+
+    let points = 0;
+
+    melds.forEach((meld) => {
+      if (meld.type === MeldEnums.TRIPLET || meld.type === MeldEnums.QUAD) {
+        const tileStr = meld.tiles[0];
+        if (winds.includes(tileStr) && tileStr === wind.valueOf()) points += HKHandMapper.WIND_MELD.points;
+      }
+    });
```

Comment:
Take player position in as well to increase points?
i.e. Wind = EAST, I'm sitting after dealer. I should get 1 point if I have SOUTH triplets
or Wind = EAST, I'm dealer. I should get 2 points if I have EAST triplets

Reply:
Good catch, also I need to factor in for concealed as well. If this can wait, I'll finish when I get back. Otherwise, I can complete it in a future PR

Reply:
didn't know these rules, so I am only looking at syntax, not the logic haha